ucb-rit/coldfront
# Add views for requesting and reviewing cluster accounts

Fixes #31

**Changes**
- Added a view for requesting a cluster account under a particular allocation.
    - Regular project users can only request accounts for themselves, while managers or PIs can request accounts for others as well.
    - The list of eligible users includes those who have signed the access agreement and who do not already have accounts.
- Added an admin view for listing cluster account requests.
- Added views for approving/denying cluster account requests.
    - In the case of approvals, the administrator should perform backend setup before approving. He/she will be prompted to provide the username and UID of the created cluster account. These will be stored in the database.
- Added a new `AllocationAttributeType` "Cluster Account Status", which takes on values such as, "Pending - Add", "Active", and "Denied".
- Added email notifications on new requests, activations, and denials.

**How to Test**
- Review the code changes and add comments as needed.
- Ensure that the test suite passes.
- Manual Testing
    - Run `python manage.py migrate`.
    - Run `python manage.py add_allocation_defaults` to create a `AllocationAttributeType` for cluster accounts statuses.
    - As a non-superuser:
        - Sign the access agreement in the User Profile.
        - Navigate to an allocation detail page.
        - Request a cluster account.
    - As a superuser:
        - Approve/deny the request.
            - For approvals, a cluster username and UID must be provided.

**Notes**
- Additional views displaying cluster accounts for a user or account need to be added separately.
- Email notifications for the entire application will be reviewed and tested on a future date.
